src/communication/communicator.hpp

```diff
@@ -101,11 +105,17 @@ class communicator {
         for (const auto& cell: gid_infos) {
             for (auto c: cell.conns) {
                 const auto i = offsets[src_domains[pos]]++;
-                connections_[i] = {c.source, c.dest, c.weight, c.delay, cell.local_group};
+                connections_[i] = {c.source, c.dest, c.weight, c.delay, cell.domain_id};
                 ++pos;
             }
         }
 
+        // Build cell partition by group for passing events to cell groups
+        domain_id_partition_.resize(num_local_groups_+1);
```

Comment:
Is this initialized to `domain_id_partiton_[0] = 0`

Reply:
Yes. See http://en.cppreference.com/w/cpp/container/vector/resize
Specifically: "If the current size is less than count additional default-inserted elements are appended".